### PR TITLE
remove test_accuracy.py

### LINKED ISSUE #19187: Modify transformer accuracy test to use Generator
Currently `models/tt_transformers/tests/test_accuracy.py` instantiates the `Transformer` model directly. Test should be modified to instantiate the model trough `Generator` class. This will allow for accuracy testing of DP/HP configurations of transformer models.

### PR DESCRIPTION
**Pull request by epam-ioannis-alexiou**
### Ticket
Close https://github.com/tenstorrent/tt-metal/issues/35813

### Problem description
file `test_accuracy.py` was removed by [PR retiring test_accuracy](https://github.com/tenstorrent/tt-metal/pull/25174/files). However, it came back from retirement somehow (maybe bad merge-from-main or rebase-onto-main by later PRs).

### What's changed
In this PR the file `test_accuracy.py` and all mentions of it are removed while the reported performance token metrics can be obtained by `ci-token-matching` pytest case of `simple_text_demo.py`
 **Changes:**
- Deletes `models/tt_transformers/tests/test_accuracy.py` (475 lines removed)
- Updates documentation in `models/tt_transformers/README.md` to reference `demo/simple_text_demo.py` instead of the deleted test file

### Checklist

- [x] [![[post-commit] all - Static checks, linters etc.](https://github.com/tenstorrent/tt-metal/actions/workflows/all-static-checks.yaml/badge.svg)](https://github.com/tenstorrent/tt-metal/actions/workflows/all-static-checks.yaml)

### CODE REVIEW

**Comment by epam-ioannis-alexiou** [comment]
@gwangTT and @ipotkonjak-tt please take a look on this PR.

**Comment by gwangTT** [comment]
Did you perform a global grep of `test_accuracy.py` to make sure there is no CI tests that rely on it? It is worth mentioning in the PR description.

File: models/tt_transformers/README.md [hunk 1]
```diff
@@ -196,7 +196,7 @@ If you are bringing up a new model that is similar to these but is not listed ab
 
 You should also watch out for:
 - RoPE encoding style. `llama3`, `yarn` and of course `none` are supported. HuggingFace models encode the complex numbers in RoPE as r1, r2, ..., i1, i2, ... whereas Meta models encode the complex numbers as r1, i1, r2, i2, ... - TTT uses a Meta-style implementation of the RoPE op and when loading a HuggingFace it will reshuffle the weights of the pre-rope attention weights to interleave their outputs in this style (see `reverse_permute` in [load_checkpoints.py](tt/load_checkpoints.py)). When _using_ TTT this happens invisibly and should not affect you, but when bringing up a new model or modifying TTT being aware of this will make some otherwise confusing things clearer.
-- Our [accuracy test](tests/test_accuracy.py) will require you to [generate some reference logits](tests/generate_reference_hf.py) and perhaps update the test to use them.
+- Our [accuracy test](demo/simple_text_demo.py) will require you to [generate some reference logits](tests/generate_reference_hf.py) and perhaps update the test to use them.
```
Comment (gwangTT) [inline models/tt_transformers/README.md]:
```suggestion
- Our [accuracy test](demo/simple_text_demo.py -k ci-token-matching) will require you to [generate some reference logits](tests/generate_reference_hf.py) and perhaps update the test to use them.
```

**Comment by epam-ioannis-alexiou** [comment]
> Did you perform a global grep of `test_accuracy.py` to make sure there is no CI tests that rely on it? It is worth mentioning in the PR description.

Yes the only reference I found was in the README.md file we commited changes.
Also there is a comment here: https://github.com/tenstorrent/tt-metal/blob/5f09dc9d7c87f5716c124c5e3a54da78ea6fd0a8/models/demos/gemma3/demo/text_demo.py#L24

**Review by gwangTT — APPROVED** [review]

**Comment by gwangTT** [comment]
> > Did you perform a global grep of `test_accuracy.py` to make sure there is no CI tests that rely on it? It is worth mentioning in the PR description.
> 
> Yes the only reference I found was in the README.md file we commited changes. Also there is comment here:
> 
> https://github.com/tenstorrent/tt-metal/blob/5f09dc9d7c87f5716c124c5e3a54da78ea6fd0a8/models/demos/gemma3/demo/text_demo.py#L24

Easy deletion. thanks, Ioannis!

**Review by ipotkonjak-tt — APPROVED** [review]

**Review by yieldthought — APPROVED** [review]

**Review by handrewsTT — APPROVED** [review]
Kill it for good this time!
